gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/OplogIndex.java

```diff
@@ -211,6 +211,12 @@ boolean checkValidIndexFile(File f) {
       // delete the existing, unreadable file
       deleteIRF(hasIrf ? "unreadable file" : "metadata missing");
     }
+    else if (recreateIndexFile) {
+      // delete the existing, file as requires recreation
+      deleteIRF(hasIrf ? "unreadable file" : "metadata missing");
+      // also delete the irf of OplogIndex
+      this.irf = null;
+    }
     this.oplog.indexesWritten.clear();
```

Comment:
See comments before. I think these changes should be part of a separate ticket/PR.